components/progress/Line.tsx

```diff
@@ -1,11 +1,72 @@
 import * as React from 'react';
 import { validProgress } from './utils';
-import { ProgressProps } from './progress';
+import { ProgressProps, ProgressGradient } from './progress';
 
 interface LineProps extends ProgressProps {
   prefixCls: string;
   children: React.ReactNode;
 }
+type StringGradients = { [percent: string]: string };
+
+/**
+ * {                             {
+ *   '0%': '#afc163',              '0%': '#afc163',
+ *   '75%': '#009900',             '25%': '#66FF00',
+ *   '50%': 'green',     ====>     '50%': '#00CC00',
+ *   '25%': '#66FF00',             '75%': '#009900',
+ *   '100%': '#ffffff'             '100%': '#ffffff'
+ * }                             }
+ */
+export const sortGradient = (gradients: StringGradients) => {
+  let tempArr = [];
+  const result: StringGradients = {};
+  for (const [key, value] of Object.entries(gradients)) {
+    const formatKey = parseFloat(key.replace('%', ''));
```

Comment:
```suggestion
    const formatKey = parseFloat(key.replace(/%/g, ''));
```

https://lgtm.com/projects/g/ant-design/ant-design/rev/pr-27de17dbd5c47fb4e5924e9865e61bcbd9a3f95d